### PR TITLE
add emptyStringTest checker

### PR DESCRIPTION
**Pull request by Quasilyte**
Suggests to replace `len(s) == 0` with `s == ""`.

See https://dmitri.shuralyov.com/idiomatic-go#empty-string-check.

Signed-off-by: Iskander Sharipov <quasilyte@gmail.com>

### CODE REVIEW

File: emptyStringTest_checker.go [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+	lenCall := astcast.ToCallExpr(cmp.X)
+	if astcast.ToIdent(lenCall.Fun).Name != "len" {
+		return
+	}
+	s := lenCall.Args[0]
+	if !typep.HasStringProp(c.ctx.TypesInfo.TypeOf(s)) {
+		return
+	}
+	zero := astcast.ToBasicLit(cmp.Y)
+	if zero.Value != "0" {
```
Comment (cristaloleg) [inline emptyStringTest_checker.go]:
```
if zero.Value == "0" {
	c.warn(cmp)
}
```

File: testdata/emptyStringTest/positive_tests.go [hunk 2]
```diff
@@ -0,0 +1,15 @@
+package checkers
+
+func badEmptyStringChecks(s string) {
+	sptr := &s
+
+	/*! replace `len(s) == 0` with `len(s) == ""` */
```
Comment (cristaloleg) [inline testdata/emptyStringTest/positive_tests.go]: `len(s) == ""`
maybe
`s == ""`
?

File: emptyStringTest_checker.go (hunk 1 above)
Comment (Quasilyte) [reply emptyStringTest_checker.go]: I usually do follow "de-indent" the happy path, unless there is compelling reason not to do so. There are already several "guards", I think it's more consistent with the way it is right now.

File: testdata/emptyStringTest/positive_tests.go (hunk 2 above)
Comment (Quasilyte) [reply testdata/emptyStringTest/positive_tests.go]: Hah! Nice catch. Thanks, I'll fix that.

**Pull request merged**